agent/lib_aws_sdk_php.c

```diff
@@ -0,0 +1,177 @@
+ * need to store it, we just need to snag it from the return value as it goes
+ * through the client initialization process.
+ */
+NR_PHP_WRAPPER(nr_aws_client_parse_class) {
+  (void)wraprec;
+
+  zval** ret_val_ptr = NULL;
+  ret_val_ptr = NR_GET_RETURN_VALUE_PTR;
+
+  NR_PHP_WRAPPER_CALL;
+
+  if (NULL != ret_val_ptr && nr_php_is_zval_valid_array(*ret_val_ptr)) {
+    /* obtain ret_val_ptr[0] which contains the service name */
+    zval* service_name
+        = nr_php_zend_hash_index_find(Z_ARRVAL_P(*ret_val_ptr), 0);
+    if (nr_php_is_zval_non_empty_string(service_name)) {
+      nr_lib_aws_sdk_php_add_supportability_service_metric(
+          Z_STRVAL_P(service_name));
+    }
+  }
+}
+NR_PHP_WRAPPER_END
+
+/*
+ * The ideal file to begin immediate detection of the aws-sdk is:
+ * aws-sdk-php/src/functions.php
+ * Unfortunately, Php8.2+ and composer autoload leads to the
+ * file being optimized directly and not loaded.
+ *
+ * Options considered:
+ *
+ * 1. for PHP8.2, and only optimizable libraries, when encountering autoload.php
+ * files, ask the file what includes it added and check against only the
+ * optimizable library. Small overhead incurred when encountering an autoload
+ * file, but detects aws-sdk-php immediately before any sdk code executes
+ * (changes needed for this are detailed in the original PR)
+ * 2. use a file that gets called later and only when AwsClient.php file file is
+ * called. It's called later and we'll miss some instrumentation, but if we're
+ * only ever going to be interested in Client calls anyway, maybe that's ok?
+ * Doesn't detect Sdk.php (optimized out) so when customers only use that or
+ * when they use it first, we will not instrument it. This only detects when a
+ * Client is called to use a service so potentially misses out on other
+ * instrumentation and misses out when customers use the aws-sdk-php but use
+ * non-SDK way to interact with the service (possibly with redis/memcached).
+ * This way is definitely the least complex and lowest overhead and less
+ * complexity means lower risk as well.
+ * 3. Directly add the wrappers to the hash map. With potentially 50ish clients
+ * to wrap, this will add overhead to every hash map lookup. Currently
+ * implemented option is 2, use the AwsClient.php as this is our main focus.
+ * This means until a call to an Aws/AwsClient function,
+ * all calls including aws\sdk calls are ignored.  Version detection will be
+ * tied to Aws/ClientResolver::_apply_user_agent which is ALWAYS called when
+ * dealing with aws clients.  It will not be computed from
+ * Aws/Sdk::__constructor which would at best be duplicate info and worst would
+ * never be ignored until a client is called.
+ */
+void nr_aws_sdk_php_enable() {
+  /* This will be used to extract the version. */
+  nr_php_wrap_user_function(NR_PSTR("Aws\\ClientResolver::_apply_user_agent"),
+                            nr_aws_version);
```

Comment:
The name of this wrapper function is not clear enough. I really prefer the code to the talking:
```suggestion
  /* This will be used to extract the version. */
  nr_php_wrap_user_function(NR_PSTR("Aws\\ClientResolver::_apply_user_agent"),
                            nr_create_aws_sdk_version_metrics);
```